src/super_gradients/training/datasets/data_formats/obb/cxcywhr.py

```diff
@@ -0,0 +1,58 @@
+import cv2
+import numpy as np
+
+
+def cxcywhr_to_poly(boxes: np.ndarray) -> np.ndarray:
+    """
+    Convert oriented bounding boxes in CX-CY-W-H-R format to a polygon format
+    :param boxes: [N,...,5] oriented bounding boxes in CX-CY-W-H-R format
+    :return: [N,...,4, 2] oriented bounding boxes in polygon format
+    """
+    shape = boxes.shape
+    if shape[-1] != 5:
+        raise ValueError(f"Expected last dimension to be 5, got {shape[-1]}")
+
+    flat_rboxes = boxes.reshape(-1, 5)
+    polys = np.zeros((flat_rboxes.shape[0], 4, 2), dtype=np.float32)
+    for i, box in enumerate(flat_rboxes):
+        cx, cy, w, h, r = box
+        rect = ((cx, cy), (w, h), np.rad2deg(r))
+        poly = cv2.boxPoints(rect)
+        polys[i] = poly
+
+    return polys.reshape(*shape[:-1], 4, 2)
+
+
+def poly_to_cxcywhr(poly: np.ndarray) -> np.ndarray:
+    shape = poly.shape
+    if shape[-2:] != (4, 2):
+        raise ValueError(f"Expected last two dimensions to be (4, 2), got {shape[-2:]}")
+
+    flat_polys = poly.reshape(-1, 4, 2)
+    rboxes = np.zeros((flat_polys.shape[0], 5), dtype=np.float32)
+    for i, poly in enumerate(flat_polys):
+        hull = cv2.convexHull(np.reshape(poly, [-1, 2]).astype(np.float32))
```

Comment:
I think I left this as a precaution. For instance, A.ElasticTransform can move (in theory) individual points such that a polygon is not convex anymore and contain self-intersections.
Additionally, a horizontal flip can change the polygon order (CW <-> CCW).

A convex hull would cover it and additionally it would return a consistent keypoints orientation (CCW) for a hull, which in turn would ensure consistent rotated box parametrization for boxes regardless of points order.